addons/base/views.py

```diff
@@ -712,6 +712,13 @@ def addon_view_or_download_file(auth, path, provider, **kwargs):
         transaction.savepoint_rollback(savepoint_id)
         if not file_node.pk:
             file_node = BaseFileNode.load(path)
+
+            if file_node.kind == 'folder':
+                raise HTTPError(httplib.BAD_REQUEST, data={
+                    'message_short': 'Bad Request',
+                    'message_long': 'File not associated with required object.'
```

Comment:
I'm not clear on what this error message means. Should it be something like "Cannot request a folder."?

Reply:
The message is just copied for consistency sake from a similar 400 at https://github.com/CenterForOpenScience/osf.io/pull/8734/files#diff-8117cabdd0bd0d9cb074b789b7ce6dc9R753, but yeah this could be clearer.

Reply:
The error seems correct in that scenario: the file isn't associated with the draft registration. I don't think it makes sense here, though.